### DISCUSSION TITLE
Try out Firefox's native Dark Mode!

### DISCUSSION THREAD

**Discussion by gunir** [Floorp-Projects/Floorp#487] [Show and tell]
You're not hearing this wrong, Firefox does have native Dark Mode, for a very long time. Just most people didn't try to turn it on and tune it up!

And Dark Reader performance is pretty bad, that's why a native Dark Mode has always been demanded by Firefox users.

How to ?
- Go to `about:config`, set:
```
browser.display.background_color #c0c0c0
browser.display.foreground_color #000000
browser.anchor_color #ffcc99
browser.display.use_system_colors false
layout.css.forced-colors.enabled false
browser.display.document_color_use 2
```

Result:
![image](https://github.com/Floorp-Projects/Floorp/assets/134402102/f68dbbe8-b61a-48a9-bd06-f88868e6c96e)

And there's many many color combination too, for example Gray Mode, Brown Mode, not only Dark Mode.

What's my message ?
If it's possible, I hope Floorp extend this feature because eventhough its performance is great, it has some flaws:
- Can't enable/disable per page/domain
- Can't set color set per page/domain, for example Dark set for DuckDuckGo, Gray set for Google, Brown set for Yahoo..

But it's very promising, I've been using it for a very long time with [Toggle Document Color (Global)](https://addons.mozilla.org/en-US/firefox/addon/toggleglobalcolors/).

There's zero performance downgrade, and it does its job.

This trick works on Mobile too,

**Comment by lazymonkey2** [comment]
Hello, how can it work on mobile?
There is no access to about:config on mobile.

**Reply by gunir** [reply]
Firefox Beta and above can access about:config, stable can't no matter what.

**Reply by lbrdt** [reply]
If you don't want to install beta, try out [Stylus](https://addons.mozilla.org/en-US/firefox/addon/styl-us/) (available on Android) and a global dark theme. Stylus is very light on resource.

Here's a how to:
https://www.jucktion.com/faster-dark-mode-on-firefox-android.

**Comment by breadbeforerice** [comment]
browser.display.background_color #c0c0c0
browser.display.foreground_color #000000
This is the exact opposite and thus not like the result.

**Reply by TwinkleByte** [reply]
Yeah. Proper config should be this.
```
browser.display.background_color #000000 default #FFFFFF
browser.display.foreground_color #c0c0c0 default #000000
browser.anchor_color #ffcc99 default #0000EE
browser.display.use_system_colors false default false
layout.css.forced-colors.enabled false default true
browser.display.document_color_use 2 default 1
```

I've provided some more images for reference. Its not that bad when we consider the fact that its not on best it can be. I would say with proper support, this can replace the resource heavy dark reader. The memory this uses is very low compared to dark reader and ultimadark that made my PC start crying noise.

This is what Arch wiki looks like using this.
![img2 ](https://github.com/user-attachments/assets/cc595564-3146-4a4c-b186-de31e6c81ad3)
A news website using this
![img2](https://github.com/user-attachments/assets/95ef9fc0-46a5-4833-a875-ed3d6310456a)

**Reply by Vistaus** [reply]
Thank you! That works better than in the OP. However, I do still have issues with images being displayed black. No such issues is Chromium and Falkon with their native dark modes. So I guess Floorps/Firefox's implementation still has ways to go.

**Comment by bphd** [comment]
> You're not hearing this wrong, Firefox does have native Dark Mode, for a very long time. Just most people didn't try to turn it on and tune it up!
> 
> And Dark Reader performance is pretty bad, that's why a native Dark Mode has always been demanded by Firefox users.
> 
> How to ?
> 
> * Go to `about:config`, set:
> 
> ```
> browser.display.background_color #c0c0c0
> browser.display.foreground_color #000000
> browser.anchor_color #ffcc99
> browser.display.use_system_colors false
> layout.css.forced-colors.enabled false
> browser.display.document_color_use 2
> ```
> 
> Result: ![image](https://private-user-images.githubusercontent.com/134402102/271811832-f68dbbe8-b61a-48a9-bd06-f88868e6c96e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU4ODk4MjIsIm5iZiI6MTcyNTg4OTUyMiwicGF0aCI6Ii8xMzQ0MDIxMDIvMjcxODExODMyLWY2OGRiYmU4LWI2MWEtNDhhOS1iZDA2LWY4ODg2OGU2Yzk2ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwOVQxMzQ1MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03N2YwNDE3ZDQwZWJjYTJkMmY5OGRlMDMzZmZjZDgzNWIxOGNhNjBkYzc2YzUwYWQzNjk2YzI2YzMyODlkNGQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.vWfFuEQ2gpI8ZmC0RrcGF4KWjWWmpiPNlrhv2jysViw)
> 
> And there's many many color combination too, for example Gray Mode, Brown Mode, not only Dark Mode.
> 
> What's my message ? If it's possible, I hope Floorp extend this feature because eventhough its performance is great, it has some flaws:
> 
> * Can't enable/disable per page/domain
> * Can't set color set per page/domain, for example Dark set for DuckDuckGo, Gray set for Google, Brown set for Yahoo..
> 
> But it's very promising, I've been using it for a very long time with [Toggle Document Color (Global)](https://addons.mozilla.org/en-US/firefox/addon/toggleglobalcolors/).
> 
> There's zero performance downgrade, and it does its job.
> 
> This trick works on Mobile too,

since a long long time FireFox has "Colors
Override ’s default colors for text, website backgrounds, and links.". But it's pretty much useless because it doesn't follow system color scheme and you have to configure manually colors. So if a website use those shades already, it will become unreadable. A Gecko fork should activate them per default

COuld you explain 

> browser.display.use_system_colors false
> layout.css.forced-colors.enabled false
> browser.display.document_color_use 2

**Comment by DealsBeam** [comment]
I prefer https://github.com/ThomazPom/Moz-Ext-UltimaDark

**Reply by bphd** [reply]
It was asked since eons that he follows browser dark/light mode and it's still not done
